Introduce custom handling of aborted jobs

keesj-exset opened the pull request:
Prior to this commit the builds that where aborted would undergo the
same handling as jobs that failed. This is not always the most desirable
behaviour. For instance the gerrit-trigger plugin allows to abort
running jobs when a new patchset is created while building. In such case
the build is aborted and a new build is started. Prior to this commit
this could result in a patchset getting a -1 verified vote causing some
confusion for the user.

If upgrading from a previous version the default behavior will remain
backwards compatible. To enable custom handing one needs to edit the
configuration and configure the build aborted default values to the
desired value of 0.

Also see https://issues.jenkins-ci.org/browse/JENKINS-14767

Original-author: Réda Housni Alaoui
Signed-off-by: Kees Jongenburger <kees.jongenburger@exset.com>

This pull request surpasses and therefore deprecates #392 by adding backwards compatibility

* Tests passed but as in the master branch ParameterModeJenkinsTest sometimes fails
 https://gist.github.com/keesj-exset/99408b19def2fbfe1d595a9e05c778a1
[INFO] Tests run: 568, Failures: 0, Errors: 0, Skipped: 0

* Tested that the upgrade of the plugin kept old behavior unchanged
* Fresh installs by default will no longer send a -1 when a build is aborted

keesj-exset commented:
@rsandell this patchset now also externalized the GerrtiTriggerDescriptor and therefore does not require changed to checkstyle.

keesj-exset commented:
@rsandell any ETA when you can have a look at the changes/review them?

keesj-exset commented:
Hi, is there anybody else who can have a look at the changes?

keesj-exset commented:
ping

rsandell commented:
Sorry, I've been away for a long time. Looking over pull requests now.

rsandell commented on src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritTriggerDescriptor.java:
```diff
@@ -0,0 +1,358 @@
+ * The Descriptor for the Trigger.
+ */
+@Extension
+//@Symbol("gerrit")
```
Needs to be put back
```suggestion
@Symbol("gerrit")
```

rsandell commented on src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritTriggerDescriptor.java:
```diff
@@ -0,0 +1,358 @@
+@Extension
+//@Symbol("gerrit")
+public final class GerritTriggerDescriptor extends TriggerDescriptor {
+
```
As I mentioned in my other [comment response](https://github.com/jenkinsci/gerrit-trigger-plugin/pull/392#discussion_r432435618) moving this was the least good option to do of those you suggested ;)
But it is a lot of code so I'll allow it :)

rsandell commented on src/main/resources/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritTrigger/config.jelly:
```diff
@@ -113,6 +115,14 @@
+                    <f:textbox name="gerritBuildAbortedVerifiedValue"
+                               value="${it.gerritBuildAbortedVerifiedValue}"
+                               default=""
+                               checkUrl="'descriptorByName/com.sonyericsson.hudson.plugins.gerrit.trigger.hudsontrigger.GerritTrigger/emptyOrIntegerCheck?value='+escape(this.value)"/>
```
Shouldn't it be 
```suggestion
                               checkUrl="'descriptorByName/com.sonyericsson.hudson.plugins.gerrit.trigger.hudsontrigger.GerritTriggerDescriptor/emptyOrIntegerCheck?value='+escape(this.value)"/>
```
?
Have you verified that all checks and list fills etc are working after the extraction of the descriptor to a separate class?

keesj-exset replied:
No, I did not do this and will perform additional testing in this area

keesj-exset replied:
No, I tested the changes and the verifiers still work as previously. The descriptorByName will still calls the static getDescriptor method from GerritTrigger hence no change is needed

rsandell commented on src/test/java/com/sonyericsson/hudson/plugins/gerrit/trigger/mock/MockGerritHudsonTriggerConfig.java:
```diff
@@ -375,7 +420,7 @@ public int getBuildScheduleDelay() {
     @Override
     public int getDynamicConfigRefreshInterval() {
-        return 30;
+        return Config.DEFAULT_DYNAMIC_CONFIG_REFRESH_INTERVAL;
```
Is this really a good idea? Or is that one `30` as well? :)

keesj-exset replied:
The following commit adds a new variable and uses it hence. yes the value is 30. It was mostly triggered to fix a check style warning. It might be possible to fix it using annotations. 

https://github.com/jenkinsci/gerrit-trigger-plugin/pull/413/commits/90a2eb84559876ffcaffc5319df7af42a52495e0

keesj-exset commented:
Hi @rsandell,

I have reviewed you review comments and checked that the form verification was indeed working. I updated you patch suggestion (The @Symbol("gerrit") ) , Added two more help files that where missing and ran the tests again. IMO the code is ready for inclusion.

keesj-exset commented:
During the test I also created a document on the setup I used for the testing.

It contains info on setting up jenkins and gerrit in .. docker... having persistence, the changes that are needed to enable the verified label and the repository used to test this patchset. 

https://github.com/keesj-exset/gerrit-trigger-plugin-test-repo/blob/master/gerrit-plugin-testing.rst

keesj-exset commented:
ping

keesj-exset commented:
Hi @rsandell,

I see other commit requests pending that might affect this patch. Is there some way for this patch to get integrated? I have done all the review and checked your comments. With kind regards

keesj-exset commented:
Is there anything I can do .. to make this happen?
With kind regards

rsandell commented:
Sorry, I don't have as much time for reviews as I used to.
